scan/src/subscriptions/ProposalSub.re

```
@@ -0,0 +1,153 @@
+type proposal_status_t =
+  | Deposit
+  | Voting
+  | Passed
+  | Rejected
+  | Failed;
+
+let parseProposalStatus = json => {
+  let status = json |> Js.Json.decodeString |> Belt_Option.getExn;
+  switch (status) {
+  | "DepositPeriod" => Deposit
+  | "VotingPeriod" => Voting
+  | "Passed" => Passed
+  | "Rejected" => Rejected
+  | "Failed" => Failed
+  | _ => Failed
```

Let's raise the error for this case. This shouldn't have happened.